### PR TITLE
More scripts

### PR DESCRIPTION
**Pull request by prafullasd**

### CODE REVIEW

File: jukebox/prior/prior.py [hunk 1]
```diff
@@ -14,10 +14,15 @@
...
-remove the vqvae pre-post processing into a separate vqvae-coder class
+Model the prior on vq codes conditioned on timing, artist, genre, lyrics and codes from levels above. 
+To condition on the timing, genre and artist, we use the LabelConditioner class
+To condition on the codes from the level above, we use the Conditioner class
+To condition on lyrics, we allow two types of priors:
+- Separate Encoder Decoder: This is the usual encoder-decoder style transformer. The encoder transformer autoregressively 
+models the lyrics, and we use its last layer to produce keys/values that are attened to by the decoder transformer
+- Single Encoder Decoder: This is a simplification where we combine them into a single model. We merge the text vocab 
+and VQ vocab into a single large vocab, and the lyric tokens and VQ tokens into a single longer sequence of tokens which 
+we autoregressively model together.
```
Comment (heewooj) [inline jukebox/prior/prior.py:25]: 👍

File: README.md [hunk 2]
```diff
@@ -88,27 +102,74 @@ Once the VQ-VAE is trained, we can restore it from its saved checkpoint and trai
...
+- Timing: 
+  - For each chunk of audio, we return the `total_length` of the song, the `offset` the current audio chunk is at and the `sample_length` of the audio chunk. We have three timing embeddings: total_length, our current position, and our current position as a fraction of the total length, and we divide the range of these values into `t_bins` discrete bins. 
+  - Pass the hps `min_duration` and `max_duration` to be the shortest/longest duration of audio files, and `t_bins` for how many bins you want to discretize timing information into. 
+
+After these modifications, to train a top-level with labels, run
+```
+mpiexec -n {ngpus} python jukebox/train.py --hps=vqvae,small_prior,all_fp16,cpu_ema --name=pretrained_vqvae_small_prior_labels \
+--sample_length=1048576 --bs=4 --aug_shift --aug_blend --audio_files_dir={audio_files_dir} \
+--labels=True --train --test --prior --levels=3 --level=2 --weight_decay=0.01 --save_iters=1000 \
+--labels_v3=True --y_bins=({artists},{genres}) --max_bow_genre_size=1 --min_duration=60.0 --max_duration=600.0 --t_bins=64
```
Comment (heewooj) [inline README.md:149]: might be useful to add that `--min_duration` `--max_duration` should be decided based on the dataset people have, and that `--min_duration` should be at least ~23.8 seconds so that there's always a context we can train on.

File: jukebox/hparams.py [hunk 3]
```diff
@@ -217,6 +217,44 @@ def setup_hparams(hparam_set_names, kwargs):
 )
 HPARAMS_REGISTRY["small_prior"] = small_prior
 
+small_single_enc_dec_prior = Hyperparams(
+    n_ctx=6144,
+    prior_width=1024,
+    prior_depth=48,
+    heads=4,
```
Comment (heewooj) [inline jukebox/hparams.py:224]: nitpick: in general, lgtm, but perhaps, 2 heads will be enough? 1B lyrics uses 2 heads as well

Comment (prafullasd) [reply jukebox/hparams.py:224]: good point, changing!
